service/dynamodb/dynamodbattribute/decode.go

```diff
@@ -454,6 +455,18 @@ func (d *Decoder) decodeString(s *string, v reflect.Value, fieldTag tag) error {
 		return d.decodeNumber(s, v)
 	}
 
+	// To maintain backwards compatibility with ConvertFrom family of methods which
+	// converted strings to time.Time structs (using JSON encoding/decoding)
+	if t, ok := v.Interface().(time.Time); ok {
+		b := []byte(*s)
+		err := t.UnmarshalJSON(b)
```

Comment:
Yes I know, but this isn't calling any Json serialization package. The MarshalJSON and UnmarshalJSON methods belong to time. I was going to simply use Parse and Format with RFC3339, but to maintain compatibility with existing implementations (which use a slight variation) it seemed more sensible to make use of these methods. Take a look at the source for time.MarshalJSON; there is no reliance on encoding/json. They are simply methods which format and parse a time string.